### ISSUE TITLE
ClasspathResourceLoaderBackend holds all resources in memory

### ISSUE THREAD

**Issue by mightyguava** [cashapp/misk#1191]
I noticed this debugging an unrelated issue. I'm not really familiar with how Java resources work, but based on the heap dump, `ClasspathResourceLoaderBackend` has a fairly big treemap (42MB) that it's keeping in memory. I think `classPath.resources.associateBy` is causing it to eagerly load data that's lazily instantiated? And then it keeps all the resources in memory because it has that tree map.

![image](https://user-images.githubusercontent.com/2769275/64621007-e7e9a800-d3b2-11e9-82e0-9568a3c0574c.png)

**Comment by mightyguava** [comment]
cc: @swankjesse

**Comment by swankjesse** [comment]
Yeah, I was a bit reluctant when I added that. And I also thought that the real problem is big .jar files with too many dependencies.

Should probably fix both problems.
